### DISCUSSION TITLE
PaddleOCR export_model出错

### DISCUSSION THREAD

**Discussion by OeasyC** [PaddlePaddle/PaddleOCR#13928] [Q&A]
AttributeError: `np.sctypes` was removed in the NumPy 2.0 release. Access dtypes explicitly instead.
![image](https://github.com/user-attachments/assets/b3ca7def-d206-477d-9cb9-8bc8c8276cca)
2.7.0版本训练效果还不错，acc可以到达0.9，python端推理顺利，但是导出模型一直会报错numpy版本问题。实际安装numpy<2.0。
2.7.0可以训练，无法导出。
2.7.0使用的预训练模型：en_PP-OCRv3_rec_train

2.8.0训练和导出都可以实现，但是训练过程中acc一直为0。
![image](https://github.com/user-attachments/assets/f05cfc58-0032-4b62-aaee-f70784dada92)
2.8.0使用的预训练模型：en_PP-OCRv4_rec_train

2.7.0和2.8.0数据集一样，2.7.0训练顺利，无法导出。2.8.0训练acc始终为0，会保存best_model且可以顺利导出。
2.7.0训练速度>2.8.0训练速度。

**Comment by GreatV** [comment]
试试main分支，2.8版本可能有训练精度的问题。

**Reply by OeasyC** [reply]
好，我现在验证一下。

**Comment by GreatV** [comment]
或者试试在2.7指定一下numpy版本

**Reply by OeasyC** [reply]
> 或者试试在2.7指定一下numpy版本

![image](https://github.com/user-attachments/assets/55ceddeb-2628-48bc-a800-7423b42c6287)
现在看好像并不是2.7.0或者2.8.0版本的区别。目前我使用2.7.0和2.8.0训练v3，acc和loss是会逐步优化的。
训练完成之后我再导出试一下。

**Reply by OeasyC** [reply]
看现象像是v3和v4导致的acc=0，loss=null

**Reply by OeasyC** [reply]
> 或者试试在2.7指定一下numpy版本

这个我尝试过卸载numpy并重新安装，指定numpy版本为1.26.4，实际运行下来还是会报错numpy>2.0的错误。请问有推荐的numpy版本吗

**Reply by OeasyC** [reply]
> 或者试试在2.7指定一下numpy版本

我用main分支训练和预测都没有问题。导出总是各种错误。一会是none无法拼接，一会是纬度不对。按照网上更新cuda版本也是会报错🤕

**Reply by GreatV** [reply]
试试回退paddle版本，paddle3.0目前支持可能不太完善。

**Comment by UserWangZz** [comment]
我看第一张图，使用的是C盘里python环境的numpy，这可能是导致numpy版本报错的原因

**Reply by OeasyC** [reply]
> 我看第一张图，使用的是C盘里python环境的numpy，这可能是导致numpy版本报错的原因

检查了一下，C盘numpy是1.26.4 ，paddle环境numpy是1.24.4，满足numpy<2.0.
请问还有其他的排查方向嘛？☺

**Reply by UserWangZz** [reply]
那就有点奇怪了

**Reply by CSaq** [reply]
我也有同样的问题
